### ISSUE TITLE
gtkui: Gtk-CRITICAL warnings when resizing album art column with the listview scrolled to the end

### ISSUE THREAD

**Issue by cboxdoerfer** [Alexey-Yakovenko/deadbeef#1391]
1. Open deadbeef from the console
2. Use a playlist with some groups and album art column
3. Scroll to the end of the playlist
4. Resize album art column

Now the console gets flooded with ```Gtk-CRITICAL **: gtk_widget_queue_draw_area: assertion 'height >= 0' failed``` error messages.

@Lithopsian, this bug was introduced in 5c0439c5e77a1d4557ca0a27ec615bc4083a2f6d

**Comment by Lithopsian** [comment]
I can't reproduce this.  I suspect that you're triggering an assertion that doesn't even exist in my version of GTK3.  The asynchronous setting of the scrollbar and scrollpos, that causes the jumping in the previous bug, can cause gdk_window_scroll() to try and scroll past the end of the scrollbar which might be the cause.  I'll submit a fix for the other bug and see if this message goes away as well.

**Comment by Lithopsian** [comment]
I still don't know what is happening here.  I don't get the assertion obviously, but I also can't trap any negative heights going to the gtk_widget_queue_draw_area() calls.  I've added an extra check to guarantee this since what used to be a no-op has been turned into an error (in GTK 3.16?).

Do you know what line is causing the assertion?  It is probably related to the scrollbar range being set beyond the upper limit, which happens routinely in the test case you describe.  This should get clamped automatically, and it does in all my testing, but maybe not in some versions or some situations.  Out of range values sneaking through to the value-changed handler shouldn't cause this problem, but it is hard to see what else could be happening.

**Comment by cboxdoerfer** [comment]
@Lithopsian, apparently the line that's causing this is https://github.com/Alexey-Yakovenko/deadbeef/blob/master/plugins/gtkui/ddblistview.c#L1091

```group_height``` sometimes gets really small there (e.g. -1000 and less)

**Comment by Lithopsian** [comment]
Thanks, that's helpful.  I've reproduced this in two situations:

1. When switching tabs, the playlist height and scroll position change completely, but the scrollbar signal handler interprets it as a scroll that can be negative.

2. With a very large number of playlists when quickly decreasing the size of the album art column, the reduction in the size of the playlist may be greater than the change in the scroll position.  The calculation ignores any change in the scrollbar configuration, so it may give a negative draw height.

The PR I submitted for issue #1392 includes a check for this condition so it should also fix this problem.
